com.unity.netcode.gameobjects/Runtime/Core/NetworkObject.cs

```diff
@@ -478,14 +480,18 @@ public void SpawnAsPlayerObject(ulong clientId, bool destroyWithScene = false)
         }
 
         /// <summary>
-        /// Despawns this GameObject and destroys it for other clients. This should be used if the object should be kept on the server
+        /// Despawns the <see cref="GameObject"/> of this <see cref="NetworkObject"/> and sends a destroy message for it to all connected clients.
         /// </summary>
+        /// <param name="destroy">(true) the <see cref="GameObject"/> will be destroyed (false) the <see cref="GameObject"/> will persist after being despawned</param>
         public void Despawn(bool destroy = false)
         {
-            NetworkManager.SpawnManager.DespawnObject(this, destroy);
+            // An edge case scenario can occur that will throw an exception due to the fact that SpawnManager is null.
+            if (NetworkManager.SpawnManager != null)
```

Comment:
Interesting, does this happen when we spawn / despawn something super early before we finish the NM initialization?

Reply:
Typically it happens during the shutdown period and (if memory serves) was specific to the editor or any form of "un-graceful" exit (i.e. killing the application not leaving room for a proper disconnection and cleanup).  Really, I am hoping we could incorporate Application.WantsToQuit into the entire shutdown process that would let us handle some of the "pre-shutdown" maintenance before we shutdown.

Reply:
ugh, I'd rather not hide this behing a safety check and log error out loud, if not throw an exception here.
if there's an order-of-operations issue plaguing and effecting other places, I think we need to fix the root cause, not hide/defend it like this.

Reply:
If anything we should throw a ticket in the backlog to come back and look at it because to @MFatihMAR point there is likely some underlying issue here where we are still processing things when we are shutting down which doesn't seem like the right thing.

Reply:
So, this can happen during unit tests... which will throw an exception...which will fail the test.
It would be great to move the priority up a bit on handling the shutdown sequence.
Application.WantsToQuit could handle destroying things so we don't run into these types of odd scenarios...if the application is going to quit then there should be a "no holds barred" on destroying anything that has a NetworkObject component and this would happen before the NetworkManager being destroyed.

Reply:
I just re-checked the edge case scenarios with the most recent merges and this issue appears to no longer be valid.
The check for a null SpawnManager no longer exists and so I am going to mark this as resolved since the topic was focused around that check.  👍